src/apps/data_movie.rb

```diff
@@ -0,0 +1,39 @@
+require 'json'
+require_relative '../movie'
+require_relative '../source'
+
+module PreserveMovie
+  def load_movie
+    data = []
+    file = './data/movie.json'
+    if File.exist?(file)
+      JSON.parse(File.read(file)).each do |movie|
+        data.push(Movie.new(movie['silent'], movie['publish_date']))
+      end
+    else
+      File.write(file, [])
+    end
+    data
+  end
+
+  def load_source
+    data = []
+    file = './data/source.json'
+    if File.exist?(file)
+      JSON.parse(File.read(file)).each do |source|
+        data.push(Source.new(source['name']))
+      end
+    else
+      File.write(file, [])
+    end
+    data
+  end
+
+  def create_movie
+    data = []
+    @movies.each do |movie|
+      data.push({ silent: movie.silent, publish_date: movie.publish_date })
+    end
+    File.write('./data/movie.json', JSON.generate(data))
+  end
```

Comment:
- [ ] OPTIONAL: Your movies are saved in the movie.json file as expected but I would like to suggest to you if you can use these formatting options to generate JSON data in a more open format, using whitespace.

```suggestion
  def create_movie
    data = []
    @movies.each do |movie|
      data.push({ silent: movie.silent, publish_date: movie.publish_date })
    end
    opts = {
       array_nl: "\n",
       object_nl: "\n",
       indent: '  ',
       space_before: ' ',
       space: ' '
    }
   File.write('./data/movie.json', JSON.generate(data, opts))
  end
```